### ISSUE TITLE
Did add a new worker node need to restart PAI service?

### ISSUE THREAD

**Issue by xulzee** [microsoft/pai#4122]
**Organization Name**:Nanjing University

**Short summary about the issue/question**: I want to add a new worker node to PAI.

**Brief what process you are following**: I created machine-list.yaml and added it completely as follows:

machine-list:
  - docker-data: "/var/lib/docker"
    hostip: "192.168.1.212"
    hostname: "vision-1080-1"
    k8s-role: "worker"
    machine-type: "gtx1080-1-node"
    nodename: "192.168.1.212"
    pai-worker: "true"
    password: "*********"
    ssh-port: "22"
    username: "pai"

Then I ran :
Python paictl.py machine add - P / path / to / cluster configuration / dir - L machine list.yaml

It ran successfully, but I went back to the platform to see that I didn't immediately add nodes. Then I tried to modify layout.yaml, push config, and restart the service, which seemed to work.

So when I add a new machine, do I have to restart the service (which brings a lot of trouble)?

**Comment by fanyangCS** [comment]
which version do you run?

**Comment by xulzee** [comment]
I use pai v0.16.0

**Comment by ydye** [comment]
Which cluster type did you deploy？

**Comment by xulzee** [comment]
Thank you very much for your reply. I'm using the yarn version

**Comment by ydye** [comment]
After adding the new node into kubernetes cluster, you should firstly update the machine into layout.yaml. And then you should push the latest config into the cluster.

After the config is pushed into the cluster, you could try to run the following command to update the machine lable and host list for yarn.
```
sudo ./paictl.py service start -n cluster-configuration
```

**Comment by zeyu-hello** [comment]
Why new added worker node don't need to specify below param?
```
machine-sku:
  GENERIC:
    mem: 128G
    gpu:
      type: geforce2080ti
      count: 8
    cpu:
      vcore: 16
    os: ubuntu16.04
```

**Comment by xulzee** [comment]
Hey@ydye @zeyu-hello , thank you very much for your help. I tried it. I used the following four steps to add a node without restarting all services.

first, created machine-list.yaml and run following command
`Python paictl.py machine add - P / path / to / cluster configuration / dir - L machine list.yaml`

seconed,  update the machine into layout.yaml

thired,  push the latest config into the cluster
`python paictl.py  config push -p ~/pai-config/`

fourth, run these command
`python paictl.py service stop -n cluster-configuration`
`python paictl.py service start -n cluster-configuration`
